### PR TITLE
Remove telemetry except of messages, which may indicate a bug in our code

### PR DESCRIPTION
**Pull request by jansorg**
Closes https://github.com/getappmap/appmap-intellij-plugin/issues/424
Closes https://github.com/getappmap/appmap-intellij-plugin/issues/398

This PR removes most of the telemetry message. Please refer to [this comment](https://github.com/getappmap/appmap-intellij-plugin/issues/398#issuecomment-1721241742) for a list of the removed messages.

### CODE REVIEW

File: plugin-gradle/src/main/java/appland/execution/AppMapGradleAgentRunner.java [hunk 1]
```diff
@@ -34,11 +34,7 @@ public boolean canRun(@NotNull String executorId, @NotNull RunProfile profile) {
     @Override
     public void execute(@NotNull ExecutionEnvironment environment) throws ExecutionException {
         verifyJdk(environment);
-        try {
-            findDelegate().execute(environment);
-        } finally {
-            AppMapJvmExecutor.sendTelemetry(environment.getRunProfile());
-        }
+        findDelegate().execute(environment);
```
Comment (ahtrotta) [inline plugin-gradle/src/main/java/appland/execution/AppMapGradleAgentRunner.java:37]: Why is this no longer in a `try` block? If it raises an exception, is it a problem that we are no longer catching it?

File: plugin-java/src/main/java/appland/execution/AbstractAppMapJavaAgentRunner.java [hunk 2]
```diff
@@ -27,11 +27,7 @@ public abstract class AbstractAppMapJavaAgentRunner extends DefaultJavaProgramRu
     @Override
     public void execute(@NotNull ExecutionEnvironment environment) throws ExecutionException {
         verifyJdk(environment);
-        try {
-            super.execute(environment);
-        } finally {
-            AppMapJvmExecutor.sendTelemetry(environment.getRunProfile());
-        }
+        super.execute(environment);
```
Comment (ahtrotta) [inline plugin-java/src/main/java/appland/execution/AbstractAppMapJavaAgentRunner.java:30]: As above, why is this no longer in a `try` block?

File: plugin-core/src/main/java/appland/webviews/findings/FindingsOverviewEditor.java [hunk 3]
```diff
@@ -53,37 +51,6 @@ protected void setupInitMessage(@Nullable List<ScannerFinding> initData, @NotNul
...
-
-            uniqueImpactDomainCounts.forEach((domain, count) -> {
-                if (count > 0) {
-                    event.metric("num_findings.num_unique_" + domain.getJsonId().toLowerCase(), count.doubleValue());
-                }
-            });
-
-            return event;
-        });
     }
```
Comment (ahtrotta) [inline plugin-core/src/main/java/appland/webviews/findings/FindingsOverviewEditor.java:54]: It looks like this is now an empty method, should we just remove the method entirely?

**Review by ahtrotta — COMMENTED** [review]
Are there automated tests that check if we are sending error telemetry? If not, could you add tests for that?

File: plugin-gradle/src/main/java/appland/execution/AppMapGradleAgentRunner.java (hunk 1 above)
Comment (jansorg) [reply plugin-gradle/src/main/java/appland/execution/AppMapGradleAgentRunner.java:37]: Try-finally means that exceptions are passed to the caller (i.e. they are not caught), but the finally code is always executed, even when an exception was thrown by the try block. 
we don‘t have code in finally anymore, so it was removed.

Comment (ahtrotta) [reply plugin-gradle/src/main/java/appland/execution/AppMapGradleAgentRunner.java:37]: Okay, makes sense, thank you!

**Comment by jansorg** [comment]
There are no tests for the remaining messages:
- `run_config:incompatible_jdk`
- `authentication:failed`
- `open_api:failure`

An incompatible JDK setup can't be tested well. Testing the remaining two messages would require a mock OAuth server setup and a mocked AppMap CLI, which would take considerable time. Could you please open a new issue if you'd like tests?

**Review by ahtrotta — APPROVED** [review]
I think if the tests are going to require that much effort then we can hold off on them for now. This looks good, thanks!

**Pull request merged**